### ISSUE TITLE
[clang-format] wrong formatting when C++ keywords are used as variable name in C

### ISSUE THREAD

**Issue by kleisauke** [llvm/llvm-project#59188]
For example:
```c
void test() {
  int *p;
  
  int and, and_eq, bitand, bitor, compl, not, not_eq, or, or_eq, xor,
      xor_eq = 1;

  p = &and;
  p = &and_eq;
  p = &bitand;
  p = &bitor;
  p = &compl;
  p = &not;
  p = &not_eq;
  p = &or;
  p = &or_eq;
  p = &xor;
  p = &xor_eq;
}

int main() {
  int *and, *and_eq, *bitand, *bitor, *compl, *not, *not_eq, *or, *or_eq,
      *xor, *xor_eq = 0;
}
```

Is incorrectly formatted as:
```c
void test() {
  int *p;

  int and, and_eq, bitand, bitor, compl, not, not_eq, or, or_eq, xor,
      xor_eq = 1;

  p = &and;
  p = & and_eq ;
  p = &bitand;
  p = & bitor ;
  p = &compl ;
  p = &not ;
  p = & not_eq ;
  p = & or ;
  p = & or_eq ;
  p = &xor;
  p = & xor_eq ;
}

int main() {
  int *and, * and_eq, *bitand, * bitor, *compl, *not, * not_eq, * or, * or_eq,
      *xor, * xor_eq = 0;
}
```
(note that `and`, `bitand`, `compl`, `not` and `xor` is somehow unaffected)

Tested with the latest revision (commit 409f0dc4a420db1c6b259d5ae965a070c169d930 at the time of writing):
```console
$ clang-format --version
clang-format version 16.0.0 (https://github.com/llvm/llvm-project.git 409f0dc4a420db1c6b259d5ae965a070c169d930)
$ clang-format -style=llvm test.c
```

(For the same reason, the [`lldb/test/API/lang/c/cpp_keyword_identifiers/main.c`](https://github.com/llvm/llvm-project/blob/409f0dc4a420db1c6b259d5ae965a070c169d930/lldb/test/API/lang/c/cpp_keyword_identifiers/main.c) file is probably excluded for clang-format)

**Comment by rymiel** [comment]
similar to https://github.com/llvm/llvm-project/issues/25697 and https://github.com/llvm/llvm-project/issues/43678 and https://github.com/llvm/llvm-project/issues/53173

**Comment by kleisauke** [comment]
> similar to #25697 and #43678 and #53173

It looks like the first two issues are duplicates of each other and have already been fixed with commit https://github.com/llvm/llvm-project/commit/d1aed486efc6d35a81ca4acbabb4203c4b91cda9, the last issue concerns only the `interface` keyword.

Issue #46915 (and #56249) are also similar here (i.e. it adds a unnecessary space), these were fixed with commit https://github.com/llvm/llvm-project/commit/dbc4d281bd6954362ccfc0747893ceaae842671b.

I responded to the linked issues that did not include references to these commits. If necessary, I could split this issue into separate ones, but I think in terms of code the change is happening in the same area.

**Comment by rymiel** [comment]
Sorry, the issue description just sounded familiar so I made a quick search, not checking if they were fixed. I'll close those ones.

**Comment by Tamaranch** [comment]
Another buggy keyword is `new`: for example clang-format outputs
```
int maximized = new &STATE_MAXIMIZED;
```
instead of
```
int maximized = new & STATE_MAXIMIZED;
```

**Comment by owenca** [comment]
> Another buggy keyword is `new`: for example clang-format outputs
> 
> int maximized = new &STATE_MAXIMIZED;
> instead of
> 
> int maximized = new & STATE_MAXIMIZED;

Please split it to another issue.

**Comment by Tamaranch** [comment]
> Please split it to another issue.

Isn't this also fixed by #128287? There doesn't seem to be an easy way to test clang-git on arch linux unfortunately, and I can't build the whole thing myself.

**Comment by owenca** [comment]
No, because `new` is not one of the C++ [alternative operator](https://en.cppreference.com/w/cpp/language/operator_alternative#:~:text=\).-,Keywords,-and%2C%20and_eq) keywords.
